### ISSUE TITLE
Export information shared/dedicated cpus in prometheus metrics (only for new ShareIdleCpus feautre)

### ISSUE THREAD

**Issue by ppalucki** [intel/cri-resource-manager#927]
**Describe the solution you'd like**
That only make sense for new ShareIdleCpus feautre.

What we have now is:
```
balloons{balloon="foo[0]",balloon_type="foo",containers="...",cpu_class="", cpus="1-4",cpus_max="32",cpus_min="0",mems="1",tot_req_millicpu="3000"} 3
```
and I would like to see:
```
balloons{balloon="foo[0]",balloon_type="foo",containers="...",cpu_class="", 
cpus="1-4", cpus_shared="4", cpus_dedicated="1-3",
cpus_max="32",cpus_min="0",mems="1",tot_req_millicpu="3000"} 4
```

optionally this info may be just available somewhere in:
/var/lib/cri-resmgr/cache 

It would be even better if you can also count number of cpus:
```
balloons{balloon="foo[0]",balloon_type="foo",containers="...",cpu_class="", 
cpus="1-4", 
cpus_shared="4", cpus_shared_count="1", 
cpus_dedicated="1-3", , cpus_shared_count="3", 
cpus_max="32",cpus_min="0",mems="1",tot_req_millicpu="3000"} 4
```

**Why this is needed**
help to debug which cpus are shared/dedicated in case of share idle cpus feature

@askervin

**Comment by askervin** [comment]
Thanks @ppalucki , this is very nice idea.

You can try out this feature in my 5LT_prototype_marquiz_allocator branch, commit d4a80212d7e34b4af273e2a401ad95da552f0240.

The simplest way to think about the ShareIdleCpus is probably this: this feature does not change the size of the balloons. They can be created, destroyed, inflated and deflated exactly as without this feature. The only difference is that after every creation/destruction/inflation/deflation the policy checks which CPUs are idle and (re)shares those CPUs to workloads in all balloons that are interested in extra CPUs.

Therefore I implemented this metrics feature with slightly different naming, in order to keep the meaning of "cpus" the same as before: they are the (dedicated) CPUs of a balloon. Now the metrics of every balloon include:
- `cpus`: dedicated CPUs of the balloon, cpuset format
- `cpus_count`: the number of dedicated CPUs of the balloon
- `sharedidlecpus`: extra CPUs for workloads in the balloon from the ShareIdleCpus feature, cpuset format
- `shareidlecpus_count`: the number of the extra CPUs
- `cpus_allowed`: all CPUs that workloads in the balloon are allowed to use (now union of cpus and sharedidlecpus), cpuset format
- `cpus_allowed_count`: the number of all CPUs that workloads in the balloon are allowed to use.

@fmuyassarov , I think we can build ShareIdleCpus test automation with these.

**Comment by fmuyassarov** [comment]
Thank you @askervin . That simplifies new tests a lot. As discussed earlier, I will update my verifications based on these.

**Comment by fmuyassarov** [comment]
@askervin What do you think about adding NUMA node id as well?

**Comment by askervin** [comment]
Interesting idea, @fmuyassarov .... maybe we could indeed have balloon topology fields like:
- `topo_packages="0"`
- `topo_dies="0,1"
- `topo_numas="1,2,3"
...which would work fine without topo_ prefix, too. Or alternatively single topology field that tells which packages/dies/numas balloon's CPUs are located into Like:
- `topo="0;0,1;1,2,3"`

Opinions? @ppalucki ?

**Comment by ppalucki** [comment]
yes, the topo information would be great and I prefer the first option 

I know that initial that was my idea, but I think we're starting to abuse the prometheus format :) a little, but yes I/m definitely for it

**Comment by askervin** [comment]
Fixed in PR #947
